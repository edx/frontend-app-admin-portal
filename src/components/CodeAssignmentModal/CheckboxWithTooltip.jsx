import React from 'react'
import ReduxFormCheckbox from '../ReduxFormCheckbox';
import IconWithTooltip from '../IconWithTooltip';

const CheckboxWithTooltip = ({ className, icon, altText, tooltipText, ...props}) => {
  return (
    <div className={className}>
      <ReduxFormCheckbox {...props} />
      <IconWithTooltip
        icon={icon}
        altText={altText}
        tooltipText={tooltipText}
      />
    </div>
  );
};

export default CheckboxWithTooltip;
