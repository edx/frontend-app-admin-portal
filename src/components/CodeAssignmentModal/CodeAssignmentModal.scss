.modal {
    .modal-dialog {
        max-width: 650px;
    }

    .modal-title {
        span {
            font-weight: 600;
        }
    }

    .modal-body {
        .field-group {
            p {
                font-weight: 600;
            }
        }

        p {
            margin-bottom: 6px;
        }
    }
}
