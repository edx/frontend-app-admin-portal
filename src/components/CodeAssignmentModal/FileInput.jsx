import React from 'react';
import PropTypes from 'prop-types';
import { Icon, ValidationMessage, Variant } from '@edx/paragon';

class FileInput extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      fileName: null,
    };

    this.fileInputRef = React.createRef();

    this.handleOnChange = this.handleOnChange.bind(this);
  }

  handleFileReaderOnLoad(fileReader) {
    const { result: fileContents } = fileReader;

    if (fileContents) {
      const emails = fileContents.split('\n');
      console.log(emails);
    }
  }

  handleOnChange() {
    const ref = this.fileInputRef && this.fileInputRef.current;

    if (ref) {
      const fileReader = new FileReader();

      const file = ref.files[0];
      const { name: fileName } = file;

      fileReader.onload = () => this.handleFileReaderOnLoad(fileReader);
      fileReader.readAsText(file);

      this.setState({
        fileName,
      });
    }
  }

  render() {
    const { fileName } = this.state;
    const {
      id,
      label,
      accept,
      description,
      disabled,
      required,
      meta: { touched, error },
    } = this.props;

    return (
      <div className="form-group">
        <label
          className="d-block"
          htmlFor={id}
        >
          {label}
        </label>
        <label
          className="btn btn-outline-primary"
          htmlFor={id}
        >
          <Icon className={['fa', 'fa-upload', 'mr-2']} />
          {fileName || 'Choose a file'}
        </label>
        <input
          ref={this.fileInputRef}
          className="sr-only"
          id={id}
          type="file"
          accept={accept}
          disabled={disabled}
          required={required}
          onChange={this.handleOnChange}
        />
        <ValidationMessage
          id={`validation-${id}`}
          isValid={!(touched && error)}
          invalidMessage={error}
          variant={{
            status: Variant.status.DANGER,
          }}
        />
        {description && (
          <small className="form-text" id={`description-${id}`}>
            {description}
          </small>
        )}
      </div>
    );
  }
}

FileInput.defaultProps = {
  accept: undefined,
  description: undefined,
  disabled: false,
  required: false,
};

FileInput.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool.isRequired,
    error: PropTypes.string.isRequired,
  }).isRequired,
  accept: PropTypes.string,
  description: PropTypes.string,
  disabled: PropTypes.bool,
  required: PropTypes.bool,
};

export default FileInput;
