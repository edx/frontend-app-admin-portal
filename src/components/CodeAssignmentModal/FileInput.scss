.file-input {
    .invalid-feedback {
        display: block;
    }
}
