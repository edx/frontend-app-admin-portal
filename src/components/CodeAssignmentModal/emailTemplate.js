const emailTemplate = `Your learning manager has provided you with a new access code to take a course at edX. You may redeem this code for {{REDEMPTIONS_REMAINING}} # of courses.

edX Login: {{USER_EMAIL}}
Enrollment URL: {{ENROLLMENT_URL}}
Access Code: {{CODE}}
Expiration Date: {{EXPIRATION_DATE}}

You may go directly to the Enrollment URL to view courses that are available for this code or you can insert the access code at checkout under “coupon code” for applicable courses.

For any questions, please reach out to your Learning Manager.
`;

export default emailTemplate;
