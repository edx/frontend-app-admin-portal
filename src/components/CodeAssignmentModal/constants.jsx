import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
import { MODAL_TYPES } from '../EmailTemplateForm/constants';
import { EMAIL_TEMPLATE_FIELDS } from '../EmailTemplateForm';
import CheckboxWithTooltip from '../ReduxFormCheckbox/CheckboxWithTooltip';

export const ASSIGNMENT_ERROR_TITLES = {
  [MODAL_TYPES.assign]: 'Unable to assign codes',
  [MODAL_TYPES.save]: 'Unable to save template',
};
export const EMAIL_TEMPLATE_NUDGE_EMAIL_ID = 'enable-nudge-emails';

export const ASSIGNMENT_MODAL_FIELDS = {
  ...EMAIL_TEMPLATE_FIELDS,
  [EMAIL_TEMPLATE_NUDGE_EMAIL_ID]: {
    name: EMAIL_TEMPLATE_NUDGE_EMAIL_ID,
    id: EMAIL_TEMPLATE_NUDGE_EMAIL_ID,
    component: CheckboxWithTooltip,
    className: 'auto-reminder-wrapper',
    icon: faInfoCircle,
    altText: 'More information',
    tooltipText: 'edX will remind learners to redeem their code 3, 10, and 19 days after you assign it.',
    label: 'Automate reminders',
  },
};
