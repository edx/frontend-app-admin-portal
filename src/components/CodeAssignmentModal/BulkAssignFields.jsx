import React from 'react';
import { Field } from 'redux-form';

import H3 from '../H3';
import TextAreaAutoSize from './TextAreaAutoSize';
import FileInput from './FileInput';

const BulkAssignFields = () => (
  <React.Fragment>
    <H3>Add Users</H3>
    <p className="mb-3">
      To add users, enter their email addresses below or upload a CSV file.
    </p>
    <Field
      id="email-addresses"
      name="email-addresses"
      component={TextAreaAutoSize}
      label="Email Addresses"
      description="Enter one email address per line."
      validate={() => undefined}
    />
    <Field
      id="csv-file-upload"
      name="csv-file-upload"
      component={FileInput}
      label="Upload Email Addresses"
      description="The file must be a CSV containing a single column of email addresses."
      validate={() => undefined}
      accept=".csv"
    />
  </React.Fragment>
);

export default BulkAssignFields;
