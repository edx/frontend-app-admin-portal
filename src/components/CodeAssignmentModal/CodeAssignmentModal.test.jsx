import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';

import CodeAssignmentModal from './index';
// import { Form } from 'redux-form';
// import BulkAssignFields from './BulkAssignFields';
// import IndividualAssignFields from './IndividualAssignFields';

// required to test Paragon modal component
ReactDOM.createPortal = node => node;

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  table: {
    'coupon-details': {
      loading: false,
      results: [],
    },
  },
  csv: {
    'coupon-details': {},
  },
});

// const initialModalData = {
//   id: 1,
//   title: 'Test Coupon',
//   start_date: '2018-06-01T12:00:00Z',
//   end_date: '2019-02-01T12:00:00Z',
//   has_error: false,
//   num_unassigned: 0,
//   num_uses: 0,
//   max_uses: 1,
// };

const CodeAssignmentModalWrapper = props => (
  <Provider store={store}>
    <CodeAssignmentModal
      {...props}
    />
  </Provider>
);

describe('<CodeAssignmentModal />', () => {
  let wrapper;

  describe('renders correctly', () => {
    it('with isBulkAssign as true', () => {
      const tree = renderer
        .create((
          <CodeAssignmentModalWrapper
            title="My title"
            onClose={() => {}}
            isBulkAssign
            data={
              {
                unassignedCodes: 3,
              }
            }
          />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with isBulkAssign as false', () => {
      const tree = renderer
        .create((
          <CodeAssignmentModalWrapper
            title="My title"
            onClose={() => {}}
            data={
              {
                code: 'XOKASOKDIO',
                remainingUses: 1337,
              }
            }
          />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with invalid as true', () => {
      const tree = renderer
        .create((
          <CodeAssignmentModalWrapper
            title="My title"
            onClose={() => {}}
            data={
              {
                code: 'XOKASOKDIO',
                remainingUses: 1337,
              }
            }
          />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('submitting', () => {
    const simulateSubmitClick = () => {
      wrapper.find('button').find('.btn-primary').simulate('click');
    };

    it('does not work when invalid is true', () => {
      delete ReactDOM.createPortal;
      const mockSubmit = jest.fn();
      wrapper = mount((
        <CodeAssignmentModalWrapper
          submit={mockSubmit}
          invalid
        />
      ));
      simulateSubmitClick();
      expect(mockSubmit).toBeCalledTimes(0);
    });

    it('works when invalid is false', () => {
      delete ReactDOM.createPortal;
      const mockSubmit = jest.fn();
      wrapper = mount((
        <CodeAssignmentModalWrapper
          submit={mockSubmit}
          invalid={false}
        />
      ));
      simulateSubmitClick();
      expect(mockSubmit).toBeCalledTimes(1);
    });
  });
});
