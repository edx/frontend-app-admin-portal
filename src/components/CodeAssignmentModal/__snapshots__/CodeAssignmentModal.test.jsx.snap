// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeAssignmentModal /> renders correctly with invalid as true 1`] = `
<div>
  <div
    className="fade"
    role="presentation"
  />
  <div
    className="modal js-close-modal-on-click fade"
    onClick={[Function]}
    role="presentation"
  >
    <div
      aria-labelledby="id6"
      aria-modal={true}
      className=""
      role="dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h2
            className="modal-title"
            id="id6"
          >
            <span
              className="d-block"
            >
              My title
            </span>
            <small>
              Code Assignment
            </small>
          </h2>
          <button
            aria-labelledby="paragonCloseModalButton2"
            className="btn p-1 js-close-modal-on-click"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-times js-close-modal-on-click"
              id="Icon1"
            />
          </button>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="assignment-details mb-4"
          >
            <p>
              Code: 
              XOKASOKDIO
            </p>
            <p>
              Remaining Uses: 
              1337
            </p>
          </div>
          <form
            onSubmit={[Function]}
          >
            <h3>
              Add User
            </h3>
            <div
              className="form-group"
            >
              <label
                className=""
                htmlFor="asInput7"
                id="label-asInput7"
              >
                Email Address
                <span
                  className="required"
                >
                  *
                </span>
              </label>
              <input
                aria-describedby="error-asInput7"
                aria-invalid={false}
                autoComplete="on"
                className="form-control"
                disabled={false}
                id="asInput7"
                name="email-address"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                readOnly={false}
                required={true}
                themes={
                  Array [
                    "danger",
                  ]
                }
                type="email"
                value=""
              />
              <div
                aria-live="polite"
                className="invalid-feedback"
                id="error-asInput7"
              >
                <span />
              </div>
            </div>
            <div
              className="mt-4"
            >
              <h3
                className={null}
              >
                Email Template
              </h3>
              <div
                className="form-group"
              >
                <label
                  htmlFor="email-template"
                >
                  Customize Message
                  <span
                    className="required"
                  >
                    *
                  </span>
                </label>
                <textarea
                  className="form-control"
                  id="email-template"
                  name="email-template"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  required={true}
                  style={
                    Object {
                      "height": 0,
                    }
                  }
                  value=""
                />
                <div
                  aria-live="polite"
                  className="invalid-feedback"
                  id="validation-email-template"
                >
                  <span />
                </div>
              </div>
            </div>
          </form>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            Assign Code
          </button>
          <button
            className="btn js-close-modal-on-click btn-secondary"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CodeAssignmentModal /> renders correctly with isBulkAssign as false 1`] = `
<div>
  <div
    className="fade"
    role="presentation"
  />
  <div
    className="modal js-close-modal-on-click fade"
    onClick={[Function]}
    role="presentation"
  >
    <div
      aria-labelledby="id4"
      aria-modal={true}
      className=""
      role="dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h2
            className="modal-title"
            id="id4"
          >
            <span
              className="d-block"
            >
              My title
            </span>
            <small>
              Code Assignment
            </small>
          </h2>
          <button
            aria-labelledby="paragonCloseModalButton2"
            className="btn p-1 js-close-modal-on-click"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-times js-close-modal-on-click"
              id="Icon1"
            />
          </button>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="assignment-details mb-4"
          >
            <p>
              Code: 
              XOKASOKDIO
            </p>
            <p>
              Remaining Uses: 
              1337
            </p>
          </div>
          <form
            onSubmit={[Function]}
          >
            <h3>
              Add User
            </h3>
            <div
              className="form-group"
            >
              <label
                className=""
                htmlFor="asInput5"
                id="label-asInput5"
              >
                Email Address
                <span
                  className="required"
                >
                  *
                </span>
              </label>
              <input
                aria-describedby="error-asInput5"
                aria-invalid={false}
                autoComplete="on"
                className="form-control"
                disabled={false}
                id="asInput5"
                name="email-address"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                readOnly={false}
                required={true}
                themes={
                  Array [
                    "danger",
                  ]
                }
                type="email"
                value=""
              />
              <div
                aria-live="polite"
                className="invalid-feedback"
                id="error-asInput5"
              >
                <span />
              </div>
            </div>
            <div
              className="mt-4"
            >
              <h3
                className={null}
              >
                Email Template
              </h3>
              <div
                className="form-group"
              >
                <label
                  htmlFor="email-template"
                >
                  Customize Message
                  <span
                    className="required"
                  >
                    *
                  </span>
                </label>
                <textarea
                  className="form-control"
                  id="email-template"
                  name="email-template"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  required={true}
                  style={
                    Object {
                      "height": 0,
                    }
                  }
                  value=""
                />
                <div
                  aria-live="polite"
                  className="invalid-feedback"
                  id="validation-email-template"
                >
                  <span />
                </div>
              </div>
            </div>
          </form>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            Assign Code
          </button>
          <button
            className="btn js-close-modal-on-click btn-secondary"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CodeAssignmentModal /> renders correctly with isBulkAssign as true 1`] = `
<div>
  <div
    className="fade"
    role="presentation"
  />
  <div
    className="modal js-close-modal-on-click fade"
    onClick={[Function]}
    role="presentation"
  >
    <div
      aria-labelledby="id3"
      aria-modal={true}
      className=""
      role="dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h2
            className="modal-title"
            id="id3"
          >
            <span
              className="d-block"
            >
              My title
            </span>
            <small>
              Code Assignment
            </small>
          </h2>
          <button
            aria-labelledby="paragonCloseModalButton2"
            className="btn p-1 js-close-modal-on-click"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-times js-close-modal-on-click"
              id="Icon1"
            />
          </button>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="assignment-details mb-4"
          >
            <p>
              <span
                className="detail-label mr-1"
              >
                Unassigned Codes:
              </span>
              3
            </p>
          </div>
          <form
            onSubmit={[Function]}
          >
            <h3
              className={null}
            >
              Add Users
            </h3>
            <p
              className="mb-3"
            >
              To add users, enter their email addresses below or upload a CSV file.
            </p>
            <div
              className="form-group"
            >
              <label
                htmlFor="email-addresses"
              >
                Email Addresses
              </label>
              <textarea
                className="form-control"
                id="email-addresses"
                name="email-addresses"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "height": 0,
                  }
                }
                value=""
              />
              <div
                aria-live="polite"
                className="invalid-feedback"
                id="validation-email-addresses"
              >
                <span />
              </div>
              <small
                className="form-text"
                id="description-email-addresses"
              >
                Enter one email address per line.
              </small>
            </div>
            <div
              className="file-input form-group"
            >
              <label
                className="d-block"
                htmlFor="csv-file-upload"
              >
                Upload Email Addresses
              </label>
              <label
                className="btn btn-outline-primary m-0"
                htmlFor="csv-file-upload"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-upload mr-2"
                  id="Icon1"
                />
                Choose a file
              </label>
              <input
                accept=".csv"
                className="sr-only"
                disabled={false}
                id="csv-file-upload"
                onChange={[Function]}
                required={false}
                type="file"
              />
              <div
                className=""
              >
                <div
                  aria-live="polite"
                  className="invalid-feedback"
                  id="validation-csv-file-upload"
                >
                  <span />
                </div>
              </div>
              <small
                className="form-text"
                id="description-csv-file-upload"
              >
                The file must be a CSV containing a single column of email addresses.
              </small>
            </div>
            <div
              className="mt-4"
            >
              <h3
                className={null}
              >
                Email Template
              </h3>
              <div
                className="form-group"
              >
                <label
                  htmlFor="email-template"
                >
                  Customize Message
                  <span
                    className="required"
                  >
                    *
                  </span>
                </label>
                <textarea
                  className="form-control"
                  id="email-template"
                  name="email-template"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  required={true}
                  style={
                    Object {
                      "height": 0,
                    }
                  }
                  value=""
                />
                <div
                  aria-live="polite"
                  className="invalid-feedback"
                  id="validation-email-template"
                >
                  <span />
                </div>
              </div>
            </div>
          </form>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            Assign Codes
          </button>
          <button
            className="btn js-close-modal-on-click btn-secondary"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CodeAssignmentModal /> with isBulkAssign as false 1`] = `
<div>
  <div
    className="fade"
    role="presentation"
  />
  <div
    className="modal js-close-modal-on-click fade"
    onClick={[Function]}
    role="presentation"
  >
    <div
      aria-labelledby="id4"
      aria-modal={true}
      className=""
      role="dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h2
            className="modal-title"
            id="id4"
          >
            <span
              className="d-block"
            >
              My title
            </span>
            <small>
              Code Assignment
            </small>
          </h2>
          <button
            aria-labelledby="paragonCloseModalButton2"
            className="btn p-1 js-close-modal-on-click"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-times js-close-modal-on-click"
              id="Icon1"
            />
          </button>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="assignment-details mb-4"
          >
            <p>
              Code: 
              XOKASOKDIO
            </p>
            <p>
              Remaining Uses: 
              1337
            </p>
          </div>
          <form
            onSubmit={[Function]}
          >
            <h3>
              Add User
            </h3>
            <div
              className="form-group"
            >
              <label
                className=""
                htmlFor="asInput5"
                id="label-asInput5"
              >
                Email Address
                <span
                  className="required"
                >
                  *
                </span>
              </label>
              <input
                aria-describedby="error-asInput5"
                aria-invalid={false}
                autoComplete="on"
                className="form-control"
                disabled={false}
                id="asInput5"
                name="email-address"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                readOnly={false}
                required={true}
                themes={
                  Array [
                    "danger",
                  ]
                }
                type="email"
                value=""
              />
              <div
                aria-live="polite"
                className="invalid-feedback"
                id="error-asInput5"
              >
                <span />
              </div>
            </div>
            <div
              className="mt-4"
            >
              <h3
                className={null}
              >
                Email Template
              </h3>
              <div
                className="form-group"
              >
                <label
                  htmlFor="email-template"
                >
                  Customize Message
                  <span
                    className="required"
                  >
                    *
                  </span>
                </label>
                <textarea
                  className="form-control"
                  id="email-template"
                  name="email-template"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  required={true}
                  style={
                    Object {
                      "height": 0,
                    }
                  }
                  value=""
                />
                <div
                  aria-live="polite"
                  className="invalid-feedback"
                  id="validation-email-template"
                >
                  <span />
                </div>
              </div>
            </div>
          </form>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            Assign Code
          </button>
          <button
            className="btn js-close-modal-on-click btn-secondary"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
